tracer/test/Datadog.Trace.TestHelpers/SpanTagAssertion.cs

```diff
@@ -89,9 +89,9 @@ public SpanTagAssertion<T> Matches(string tagName, T expectedValue)
                 _tags.Remove(tagName);
             }
 
-            if (!value.Equals(expectedValue))
+            if (value is null || !value.Equals(expectedValue))
```

Comment:
Ran into a null reference exception here locally with the `ProcessStart` tests

Reply:
Interesting, that seems like it _could_ be a bug, depending on what was throwing? 🤔

Reply:
When I ran the `ProcessStart` tests locally it'd throw when attempting to find the `"cmd.component", "process"` key/value [here]( https://github.com/DataDog/dd-trace-dotnet/blob/6e1167316ac9bfc4e0fcf5addf51cd4fbe62441d/tracer/test/Datadog.Trace.TestHelpers/SpanMetadataV1Rules.cs#L588C17-L588C53) and in V0

But since it didn't happen on CI I figured that I was missing something - the tag wasn't there when debugging locally either.